WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -69,6 +62,15 @@ def index
       @competitions = @competitions.select { |competition| competition.contains?(params[:search]) }
     end
 
+    unless params[:event_ids].empty?
+      @competitions = @competitions.select do |competition|
+        (competition.eventSpecs.split & params[:event_ids]).any?
```

Comment:
Oh, it works, I've written this before, but then realised that we do unnecessary work here. Look we are retriving the whole Events both in #events method and mapping the required events. So why wouldn't we just compare the ids? Am I missing something?